pyproject.toml

```
@@ -69,7 +69,7 @@ test = [
     "pre-commit>=2.9.2, <5.0",  # The hook `mypy` requires pre-commit version 2.9.2.
     "pytest-cov>=3,<7",
     "pytest-mock>=1.7.1, <4.0",
-    "pytest-xdist[psutil]~=2.2.1",
+    "pytest-xdist[psutil]>=2.2.1,<3.9.0",
```

```suggestion
    "pytest-xdist[psutil]>=2.2.1, <3.0",
```
Let's try something more conservative first.